### DISCUSSION TITLE
riverpod notifier update state doesn't work if it is not watched in a widget.

### DISCUSSION THREAD

**Discussion by Hadit1993** [rrousselGit/riverpod#2885] [Q&A]
This is my notifier for a countdown timer:
```
const timerAmount = 120;

@riverpod
class TimerNotifier extends _$TimerNotifier {
  StreamSubscription<int>? _tickerSubscription;

  @override
  int build() {
    return timerAmount;
  }

  void triggerTimer(bool shouldStart) {
// ignore: avoid_manual_providers_as_generated_provider_dependency
    final ticker = ref.watch(tickerProvider);

    _tickerSubscription?.cancel();

    if (shouldStart) {
      state = timerAmount;

      _tickerSubscription = ticker.tick(ticks: timerAmount).listen((duration) {
        tickTimer(duration);
        debugPrint("duration: $duration");
      });
    }
  }

  void tickTimer(int duration) {
    if (duration >= 0) {
      state = duration;
    } else {
      triggerTimer(false);
    }
  }
}
```

I use this code in one of my screens to trigger the timer:
` ref.read(timerNotifierProvider.notifier).triggerTimer(true);`

but I don't watch my timerNotifierProvider in this screen. 
I navigate to my timer screen to show countdown timer and watch to timerNotifierProvider but it just shows the initial timer value and state doesn't update.

if I watch  timerNotifierProvider in my previous screen this problem will be solved.

**Comment by rrousselGit** [comment]
If your notifier isn't watched/listened, then it's considered unused and will get destroyed.

Maybe you'll want something such as `@Riverpod(keepAlive: true)`

**Comment by AhmedLSayed9** [comment]
You can register an empty listener `ref.listen` to your provider at the screen where you do `triggerTimer`.
This will prevent your provider from being disposed unless you leave that screen.
